pkg/rest/coverage/swagger_analysis.go

```diff
@@ -0,0 +1,118 @@
+package coverage
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/loads"
+	"github.com/go-openapi/spec"
+)
+
+var refCache = make(map[string]int)
+
+type RequestStats struct {
+	Body         map[string]int
+	Query        map[string]int
+	ParamsHit    int
+	ParamsNum    int
+	MethodCalled bool
+	Path         string
+	Method       string
+}
+
+func getRESTApiStats(swaggerPath string, filter string) (map[string]map[string]*RequestStats, error) {
+	restAPIStats := make(map[string]map[string]*RequestStats)
+
+	document, err := loads.JSONSpec(swaggerPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mp := range document.Analyzer.OperationMethodPaths() {
+		v := strings.Split(mp, " ")
+		if len(v) != 2 {
+			return nil, fmt.Errorf("Invalid method:path pair '%s'", mp)
+		}
+		method, path := v[0], v[1]
+
+		// filter requests uri
+		if !strings.HasPrefix(path, filter) {
+			continue
+		}
+
+		if _, ok := restAPIStats[path]; !ok {
+			restAPIStats[path] = make(map[string]*RequestStats)
+		}
+
+		if _, ok := restAPIStats[path][method]; !ok {
+			restAPIStats[path][method] = &RequestStats{
+				Query:  make(map[string]int),
+				Path:   path,
+				Method: method,
+			}
+		}
+
+		addSwaggerParams(restAPIStats[path][method], document.Analyzer.ParamsFor(method, path), document.Spec().Definitions)
+	}
+
+	return restAPIStats, nil
+}
+
+func addSwaggerParams(requestStats *RequestStats, params map[string]spec.Parameter, definitions spec.Definitions) {
+	for _, param := range params {
+		switch param.In {
+		case "body":
+			requestStats.Body = make(map[string]int)
+		case "query":
+			requestStats.Query[param.Name] = 0
+		default:
+			continue
+		}
+
+		if param.Schema != nil {
```

Comment:
I figured that a schema can only be present in case of body param, right? Might be able to move this into `case "body"` and the other one into `case "query" and remove the if statement?

Reply:
I like the idea, will do it.

Reply:
I cannot do it, body params do not always have a schema reference if that's the case we still should ++ params number. I cannot remove this condition.